Kafka Upstream plugin datatypes, schema corrections to params (#2597)
Kong/docs.konghq.com: app/_hub/kong-inc/kafka-upstream/index.md

---
name: Kafka Upstream
publisher: Kong Inc.
version: 1.3-x

> # internal 0.0.2, only former version 0.0.1.

Reviewer: didn't get this part

`internal 0.0.2, only former version 0.0.1.`

same question from #2595 applies

Author: This is more of a note to self that there are plugin version issues, and that updates did not happen as they should have in the past. You'll note there is only an index file, no other versions. Need to discuss this with @eskibars and come up with a solution as to how we present and update plugin versions. Confluent only presents the most current version of a connector on their hub; perhaps we should do the same.

    - name: keepalive
      required: false
      default: "`60000`"
      value_in_examples: 60000
      datatype: integer
      description: |
         Keepalive timeout in milliseconds.
    - name: forward_method
      required: semi
      default: "`false`"
      datatype: boolean
      description: |
         Include the request method in the message. At least one of these must be true:
         `forward_method`, `forward_uri`, `forward_headers`, `forward_body`.
    - name: forward_uri
      required: semi
      default: "`false`"
      datatype: boolean
      description: |
         Include the request URI and URI arguments (i.e., query arguments) in the message.
         At least one of these must be true: `forward_method`, `forward_uri`, `forward_headers`,
        `forward_body`.
    - name: forward_headers
      required: semi
      default: "`false`"
      datatype: boolean
      description: |
         Include the request headers in the message. At least one of these must be true:
         `forward_method`, `forward_uri`, `forward_headers`, `forward_body`.
    - name: forward_body
      required: semi
      default: "`true`"
      datatype: boolean
      description: |

>          Include the request body in the message. At least one of these must be true:

Author: correct desc request body, not header

      value_in_examples: <BOOTSTRAP_SERVERS>
      urlencode_in_examples: true
      default:
      datatype: set of record elements
      description: |

>         Set of bootstrap brokers in a `{host: host, port: port}` list format.

Reviewer: same suggestion from https://github.com/Kong/docs.konghq.com/pull/2595/#discussion_r578388500 applies here

Author: Done

When encoding request bodies, several things happen:

* For requests with content-type header of `application/x-www-form-urlencoded`, `multipart/form-data`,
  or `application/json`, this plugin will pass the raw request body on the `body` attribute, but also tries
  to return a parsed version of those arguments in `body_args`. If this parsing fails, an error message is
  returned and the message is not sent.
* If the `content-type` is not `text/plain`, `text/html`, `application/xml`, `text/xml`, or `application/soap+xml`,
  then the body will be base64-encoded to ensure that the message can be sent as JSON. In such a case,
  the message has an extra attribute called `body_base64` set to `true`.

## Known issues and limitations

Known limitations:

1. There is no support for TLS.
2. There is no support for Authentication.
3. There is no support for message compression.
4. The message format is not customizable.

## Quickstart

> The following guidelines assume that both `Kong` and `Kafka` have been installed on your local machine.

Reviewer: If `Kong Enterprise` is installed on your local machine, then it already comes with `Kafka-Log` bundled with it.

Author: > If `Kong Enterprise` is installed on your local machine, then it already comes with `Kafka-Log` bundled with it.

Thanks for confirming. I know you meant kafka upstream, not log.

When encoding request bodies, several things happen:

* For requests with content-type header of `application/x-www-form-urlencoded`, `multipart/form-data`,

>   or `application/json`, this plugin will pass the raw request body on the `body` attribute, but also tries

Reviewer 2: ```suggestion
  or `application/json`, this plugin passes the raw request body in the `body` attribute, and tries
```

  to return a parsed version of those arguments in `body_args`. If this parsing fails, an error message is
  returned and the message is not sent.
* If the `content-type` is not `text/plain`, `text/html`, `application/xml`, `text/xml`, or `application/soap+xml`,
  then the body will be base64-encoded to ensure that the message can be sent as JSON. In such a case,
  the message has an extra attribute called `body_base64` set to `true`.

## Known issues and limitations

Known limitations:

1. There is no support for TLS.

> 2. There is no support for Authentication.

Reviewer 2: ```suggestion
2. There is no support for authentication.
```

3. There is no support for message compression.
4. The message format is not customizable.

## Quickstart

> The following guidelines assume that both `Kong` and `Kafka` have been installed on your local machine.

Reviewer: ```suggestion
The following guidelines assume that both `Kong` has been installed and that `Kafka Upstream` has been enabled.
```

> The following guidelines assume that both `Kong` has been installed and that `Kafka Upstream` has been enabled.

Reviewer 2: ```suggestion
The following guidelines assume that {{site.base_gateway}} is installed and the Kafka Upstream plugin is enabled.
```

> 1. Create a `kong-upstream` topic in your `Kafka` cluster:

Reviewer 2: ```suggestion
1. Create a `kong-upstream` topic in your Kafka cluster:
```
